src/interfaces/form/form.interface.ts

```diff
@@ -9,39 +9,53 @@ export interface InputComponentProps {
   inputHandler: ChangeEventHandler<HTMLInputElement>;
 }
 
-export interface FormProps {
-  [key: string]: any;
+export interface LoginFormFields {
+  email: string;
+  password: string;
+}
+export interface SignupFormFields {
+  username: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+}
+
+export type SignupFormFieldName = keyof SignupFormFields;
+
+export type LoginFormFieldName = keyof LoginFormFields;
+
+export type FormFields = SignupFormFields | LoginFormFields;
+
+export type FormFieldName = keyof FormFields;
+
+export interface FormProps<T> {
+  values: T;
   validationRules: ValidationProps;
 }
 
 export type Validator = (
   inputName: string,
-  values: FormProps,
+  values: FormFields,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   arg?: any,
 ) => string;
 
 export interface FormValidation {
   validator: Validator;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
   arg?: any;
 }
 
 export interface ValidationProps {
   [key: string]: FormValidation | FormValidation[];
 }
 
-export interface FormUIProps extends FormProps {
-  entryHeaderText: string;
-  buttonText: string;
-}
-
 export type FormTestParams = (
   testTitle: string,
-  formSetting: FormProps,
+  formSetting: FormProps<FormFields[]>,
   inputLength: number,
 ) => void;
 
-export type FormAttributes = (element: HTMLInputElement, value: string) => void;
```

Comment:
removed this and just added the argument types to the function. with the validator fn's I see the point of making a type for the function, but in this case I don't see the benefit (especially since it was just used in test file, would be good to keep separate the ones used for prod code)